src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,384 @@
+	{
+		using namespace Schema::V1_0;
+		using namespace SQLite::Builder;
+		const std::string rowIdColumnName = "rowid";
+
+		std::map<std::string, SQLite::rowid_t> result;
+
+		StatementBuilder selectStatement;
+		selectStatement.Select({ rowIdColumnName, columnName }).From(tableName).Where(columnName).In(values.size());
+
+		SQLite::Statement select = selectStatement.Prepare(connection);
+		int bindIndex = 1;
+		for (const std::string& value : values)
+		{
+			select.Bind(bindIndex, value);
+			bindIndex++;
+		}
+
+		while (select.Step())
+		{
+			result.emplace(select.GetColumn<std::string>(1), select.GetColumn<SQLite::rowid_t>(0));
+		}
+
+		return result;
+	}
+
+	void DependenciesTable::Create(SQLite::Connection& connection)
+	{
+		using namespace SQLite::Builder;
+		SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createDependencyTable_v1_3");
+
+		StatementBuilder createTableBuilder;
+		createTableBuilder.CreateTable(TableName()).BeginColumns();
+		createTableBuilder.Column(IntegerPrimaryKey());
+
+		std::vector<DependenciesTableColumnInfo> dependenciesColumns =
+		{ 
+			{ s_DependenciesTable_Manifest_Column_Name },
+			{ s_DependenciesTable_MinVersion_Column_Name },
+			{ s_DependenciesTable_PackageId_Column_Name }
+		};
+
+		for (const DependenciesTableColumnInfo& value : dependenciesColumns)
+		{
+			createTableBuilder.Column(ColumnBuilder(value.Name, Type::Int64).NotNull());
+		}
+
+		createTableBuilder.EndColumns();
+
+		createTableBuilder.Execute(connection);
+
+		StatementBuilder createPKIndexBuilder;
+		createPKIndexBuilder.CreateIndex(s_DependenciesTable_Index_Name).On(s_DependenciesTable_Table_Name).Columns({ s_DependenciesTable_Manifest_Column_Name });
+		createPKIndexBuilder.Execute(connection);
+
+		savepoint.Commit();
+	}
+
+
+	void DependenciesTable::AddDependencies(SQLite::Connection& connection, const Manifest::Manifest& manifest, SQLite::rowid_t manifestRowId)
+	{
+		SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "add_dependencies");
+
+		auto dependencies = GetDependencies(manifest, Manifest::DependencyType::Package);
+		if (!dependencies.size())
+		{
+			return;
+		}
+
+		InsertManifestDependencies(connection, manifestRowId, dependencies);
+
+		savepoint.Commit();
+	}
+
+	bool DependenciesTable::UpdateDependencies(SQLite::Connection& connection, const Manifest::Manifest& manifest, SQLite::rowid_t manifestRowId)
+	{
+		SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "update_dependencies");
+
+		const auto dependencies = GetDependencies(manifest, Manifest::DependencyType::Package);
+		const std::set<Manifest::Dependency> dependenciesSet(dependencies.begin(), dependencies.end());
+		if (!dependencies.size())
```

Comment:
What if there were dependencies but they were removed?

Reply:
see line 223 - 233 and 236